src/source.cpp

```diff
@@ -47,9 +47,20 @@ vector<unique_ptr<Source>> external_sources;
 }
 
 //==============================================================================
-// Source create implementation
+// Source implementation
 //==============================================================================
 
+Source::Source(pugi::xml_node node)
+{
+  // Check for source strength
+  if (check_for_node(node, "strength")) {
+    strength_ = std::stod(get_node_value(node, "strength"));
+  }
+
+  // Check for additional defined constraints
+  check_for_constraints(node);
```

Comment:
This method is not really accurately named. It's not just checking for constraints, but actually reading them too. We should call it `read_constraints` or similar.